### PR TITLE
Waveform display for use in the editor

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu/pull/1337
- [x] https://github.com/ppy/osu-framework/pull/1070

This PR is getting pretty lengthy, so I'm going to stop it at the point where it's visible in a test case for now, with implementation into an actual editor component coming soon.

### CODE REVIEW

File: osu.Game/Screens/Edit/Menus/EditorMenuBar.cs [hunk 1]
```diff
@@ -11,11 +11,19 @@
...
 using OpenTK;
 using OpenTK.Graphics;
+using System;
 
 namespace osu.Game.Screens.Edit.Menus
 {
     public class EditorMenuBar : OsuMenu
     {
+        /// <summary>
+        /// Invaoked when the selected mode has changed.
```
Comment (turbedi) [inline osu.Game/Screens/Edit/Menus/EditorMenuBar.cs]: Typo, should be `Invoked`

Comment (smoogipoo) [reply osu.Game/Screens/Edit/Menus/EditorMenuBar.cs]: Old commit - don't look at this for any editor changes, instead look at https://github.com/ppy/osu/pull/1337

**Comment by peppy** [comment]
missing files

**Comment by smoogipoo** [comment]
Because I didn't include the framework change this time around...

**Review by peppy — APPROVED** [review]

**Pull request merged**